proficio_sandbox/haptic_world.h

```diff
@@ -0,0 +1,161 @@
+                if (err)
+                {
+                        syslog(LOG_ERR,"%s: Could not set output buffer size.",__func__);
+                        throw std::runtime_error("(NetworkHaptics::NetworkHaptics): Ctor failed. Check /var/log/syslog.");
+                }
+
+                buflenlen = sizeof(buflen);
+                err = getsockopt(sock, SOL_SOCKET, SO_SNDBUF, (char *)&buflen, &buflenlen);
+                if (err)
+                {
+                        syslog(LOG_ERR,"%s: Could not get output buffer size.",__func__);
+                        throw std::runtime_error("(NetworkHaptics::NetworkHaptics): Ctor failed. Check /var/log/syslog.");
+                }
+                syslog(LOG_ERR,"%s: Note, output buffer is %d bytes.",__func__,buflen);
+
+                /* Set up the bind address */
+                bind_addr.sin_family = AF_INET;
+                bind_addr.sin_port = htons(port);
+                bind_addr.sin_addr.s_addr = htonl(INADDR_ANY);
+                err = bind(sock, (struct sockaddr *)&bind_addr, sizeof(bind_addr));
+                if (err == -1)
+                {
+                        syslog(LOG_ERR,"%s: Could not bind to socket on port %d.",__func__,port);
+                        throw std::runtime_error("(NetworkHaptics::NetworkHaptics): Ctor failed. Check /var/log/syslog.");
+                }
+
+                /* Set up the other guy's address */
+                their_addr.sin_family = AF_INET;
+                their_addr.sin_port = htons(port);
+                err = ! inet_pton(AF_INET, remoteHost, &their_addr.sin_addr);
+                if (err)
+                {
+                        syslog(LOG_ERR,"%s: Bad IP argument '%s'.",__func__,remoteHost);
+                        throw std::runtime_error("(NetworkHaptics::NetworkHaptics): Ctor failed. Check /var/log/syslog.");
+                }
+
+                /* Call "connect" to set datagram destination */
+                err = connect(sock, (struct sockaddr *)&their_addr, sizeof(struct sockaddr));
+                if (err)
+                {
+                        syslog(LOG_ERR,"%s: Could not set datagram destination.",__func__);
+                        throw std::runtime_error("(NetworkHaptics::NetworkHaptics): Ctor failed. Check /var/log/syslog.");
+                }
+
+
+                if (em != NULL) {
+                        em->startManaging(*this);
+                }
+        }
+
+        virtual ~NetworkHaptics() {
+                mandatoryCleanUp();
+                close(sock);
+        }
+
+protected:
+        virtual void operate() {
+                v = input.getValue();
+
+                {
+                        // send() and recv() cause switches to secondary mode. The socket is
+                        // non-blocking, so this *probably* won't impact the control-loop
+                        // timing that much...
```

Comment:
Is this something that should be tested? At least the wording of the comment should be changed before the general public can read it.